ygen/genir.go

```diff
@@ -72,7 +73,7 @@ type IROptions struct {
 // applied to the schema whilst generating the IR.
 //
 // GenerateIR returns the complete ygen intermediate representation.
-func GenerateIR(yangFiles, includePaths []string, newLangMapper NewLangMapperFn, opts IROptions) (*IR, error) {
+func GenerateIR(yangFiles, includePaths []string, langMapper LangMapper, opts IROptions) (*IR, error) {
```

Comment:
I guess the logic here is that you are using the LangMapper to store some state that is then used in its functionality? It seems like there is a risk here that you could end up with some state already being there that breaks the output -- should we have some `Reset()` function in the interface that allows us to remove all non-argument states so that callers don't risk this stale data?

Reply:
I see why you created `NewLangMapperFn` now. I actually did this to avoid writing extra boilerplate code, but I think I'll change it back.

Reply:
Actually I'm not so sure about adding it back again because if we pass in a function we're also saying we might instantiate two versions of them, which is also what we don't want the user to have to prepare for. I think the user just has to design their `LangMapper` in such a way that any deterministic ordering of calls always produces a deterministic set of naming -- i.e. we guarantee that the `LangMapper` functions are always called in a certain order. OR alternatively (which I prefer more) that we specifically say that the ordering in which methods are called is unstable and may change over time, such that the user should be prepared for this and design their langMapper in such a way such that it is robust to this, by, for example, handling naming uniquification in a later stage.

Reply:
[Added comments that LangMapper should be idempotent](https://github.com/openconfig/ygot/pull/650/commits/62eecf269f5fc76bd8a7daa56c1446cdd0f7f15c)